packages/ecmascript/src/Set.ts

```diff
@@ -49,17 +49,19 @@ export default class Set<T extends string | number> {
 		return values;
 	}
 
-	has(value): boolean {
+	public has(value: any): boolean {
 		return this.items.hasOwnProperty(value);
 	}
 
-	add(value): any {
+	public add(value: any): any {
 		this.items[value] = true;
+		this.size++;
```

Comment:
I think we need the same check for existing as in Map here.
to avoid incrementing the size when adding an already existing value.

Reply:
You're right, thanks for the node, code updated.